frontend/src/scenes/authentication/signup/signupForm/test/SignupForm.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react'
+import { Link } from 'lib/components/Link'
+import { useActions, useValues } from 'kea'
+import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
+import { signupLogic } from './signupLogic'
+import { userLogic } from '../../../../userLogic'
+import { SceneExport } from 'scenes/sceneTypes'
+import { LemonButton } from '@posthog/lemon-ui'
+import { AlertMessage } from 'lib/components/AlertMessage'
+import { IconArrowLeft } from 'lib/components/icons'
+import { SpinnerOverlay } from 'lib/components/Spinner/Spinner'
+import { SignupPanel1 } from './panels/SignupPanel1'
+import { SignupPanel2 } from './panels/SignupPanel2'
+
+export const scene: SceneExport = {
+    component: SignupForm,
+    logic: signupLogic,
+}
+
+export function SignupForm(): JSX.Element | null {
+    const { preflight } = useValues(preflightLogic)
+    const { user } = useValues(userLogic)
+    const { isSignupPanel2Submitting, signupPanel2ManualErrors, panel } = useValues(signupLogic)
+    const { setPanel } = useActions(signupLogic)
+    const [showSpinner, setShowSpinner] = useState(true)
+
+    useEffect(() => {
```

Comment:
What's the purpose of this spinner effect? Is it just aesthetic or we're waiting for something?

Reply:
Not waiting for anything. Mostly because when the net panel shows, it feels really abrupt without any sort of transition. I liked the spinner effect but can change it for something else if yo'd rather.